app/main.py

```diff
@@ -1,3 +1,65 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+
+
+from app.customer import Customer
+from app.shop import Shop, print_check
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
```

Comment:
1.Think how you can structure your code better. A good solution would be to have about 10 lines of code in this function. Here you should take json file, create customers and call necessary methods on customer to make a shop trip.
2.You should create more classes (e.g. Car, Location) they will all have some methods that you can reuse in other classes to complete this task (for example calculate car trip cost and calculate distance between two locations).
3. Add methods to your customer class - you should write methods here that will simulate a shop trip.
4. Also your tests are failing so keep working on it :)